torch/csrc/utils/tensor_dtypes.cpp

```diff
@@ -0,0 +1,93 @@
+#include <Python.h>
+#include "tensor_dtypes.h"
+#include "torch/csrc/Dtype.h"
+#include "torch/csrc/DynamicTypes.h"
+#include "torch/csrc/Exceptions.h"
+#include "torch/csrc/autograd/generated/VariableType.h"
+#include "torch/csrc/utils/tensor_types.h"
+
+namespace torch { namespace utils {
+
+static std::pair<std::string, std::string> getDtypeNames(at::ScalarType scalarType) {
```

Comment:
Is there a reason why we don't want to make those `const char*`? Remember that this will create copies of both strings, irrespectively of what you'll be doing with them (not a big deal if this is only used in initialization though).

Reply:
I just did it that way because it's only in initialization and std::strings are nicer to use.  I can change it if you want.